LearningMissionSimulation/PlaygroundVahe.cs

```diff
@@ -86,22 +92,63 @@ public void ActivateAccounts()
             }
         }
 
+        public void CreateSubjects(int subjectCount)
+        {
+            for (int i = 0; i < subjectCount; i++)
+            { 
+                Subject subject = ObjectGenerator.GenerateSubject();
+                subjectList.Add(subject);
+                subjectIdList.Add(subject.Id);
+            }
+        }
+
+        public void CreateModules(int moduleCount)
+        {
+            for (int i = 0;  i < moduleCount; i++)
+            {
+                Guid subjectId = subjectIdList[AttributeGenerator.random.Next(0, subjectIdList.Count)];
+                Module module = ObjectGenerator.GenerateModule(subjectId);
+                moduleDictionary.Add(module.Id, module);
+                moduleIdList.Add(module.Id);
+            }
+        }
+
         public void AssignModulesToInstructors()
         {
-            throw new NotImplementedException();
+            foreach (var instructor in instructorList)
+            {
+                instructor.ModuleList = GetModuleList();
+            }
         }
 
         public void AssignModulesToStudents()
         {
-            throw new NotImplementedException();
+            foreach (var student in studentList)
+            {
+                student.CompletedModuleList = GetModuleList();
+            }
         }
 
-        public void CreateClassrooms(int classroomCount)
+        List<Module> GetModuleList()
         {
-            throw new NotImplementedException();
+            List<Module> moduleList = new List<Module>();
+            int moduleCount = moduleIdList.Count;
+            int maxModuleCountLimit = 7;
+            int minModuleCountLimit = 1;
+            maxModuleCountLimit = Math.Min(moduleCount, maxModuleCountLimit);
+            minModuleCountLimit = Math.Min(moduleCount, minModuleCountLimit);
+            int count = AttributeGenerator.random.Next(minModuleCountLimit, maxModuleCountLimit);
+            for (int i = 0; i < moduleCount; i++)
```

Comment:
Your method allows to add the same method multiple times to the list. It is incorrect. You can add each module only once. Therefore, before adding you should be able to check if the list already contains that module.

Reply:
Thankk you for your comment i issues mi fixed.